src/main/java/com/github/jdramaix/client/HomePage.java

```diff
@@ -0,0 +1,128 @@
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.dom.client.DivElement;
+import com.google.gwt.dom.client.TextAreaElement;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.query.client.Properties;
+import com.google.gwt.query.client.js.JsUtils;
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.uibinder.client.UiHandler;
+import com.google.gwt.user.client.Window;
+import com.google.gwt.user.client.ui.IsWidget;
+import com.google.gwt.user.client.ui.Widget;
+
+import com.github.jdramaix.client.fileapi.Blob;
+import com.github.jdramaix.client.fileapi.Callback;
+import com.github.jdramaix.client.fileapi.FormData;
+import com.github.jdramaix.client.fileapi.XMLHttpRequest;
+
+public class HomePage implements IsWidget {
+  public interface Binder extends UiBinder<Widget, HomePage> {
+  }
+
+  private static Binder uiBinder = GWT.create(Binder.class);
+
+  @UiField
+  TextAreaElement cssContent;
+  @UiField
+  TextAreaElement resultContent;
+  @UiField
+  DivElement logContainer;
+
+  private Widget container;
+  private String uploadUrl;
+
+  public HomePage() {
+    container = uiBinder.createAndBindUi(this);
+    uploadUrl = readInitialUploadUrl();
+  }
+
+  @Override
+  public Widget asWidget() {
+    return container;
+  }
+
+  @UiHandler("convert")
+  void onConvertClicked(ClickEvent e) {
+    convert(cssContent.getValue());
+  }
+
+  private void convert(String value) {
+    Blob blob = createCssBlob(value);
+
+    FormData formData = createFormData();
+    formData.append("content", blob);
+
+    final XMLHttpRequest xhr = createXmlHttpRequest();
+
+    xhr.open("POST", uploadUrl);
+
+    xhr.setOnreadystatechange(new Callback() {
+      @Override
+      public void apply() {
+        onreadystatechange(xhr);
+      }
+    });
+
+    xhr.send(formData);
+  }
+
+  private void onreadystatechange(XMLHttpRequest xhr) {
+    if (xhr.getReadyState() == 4) {
+      Properties json = JsUtils.parseJSON(xhr.getResponseText());
+      if (xhr.getStatus() == 200) {
+        displayAnswer(json.getStr("result"));
+        displayLogs(json.getStr("logs").replaceAll("\\n", "<br>"));
+      } else {
+        Window.alert("Ooops something went wrong... Please try again!");
+        console.log(json.getStr("result"));
+      }
+
+      uploadUrl = json.getStr("nextUrl");
+    }
+  }
+
+  private void displayAnswer(String response) {
+    resultContent.setValue(response);
+  }
+
+  private void displayLogs(String logs) {
+    logContainer.setInnerHTML(logs);
+  }
+
+  private native Blob createCssBlob(String content) /*-{
+      return new Blob([content], {type: 'text/css'});
+  }-*/;
+
+  private native FormData createFormData() /*-{
+      return new FormData();
+  }-*/;
+
+  private native XMLHttpRequest createXmlHttpRequest() /*-{
+      return new XMLHttpRequest();
+  }-*/;
+
+  private native String readInitialUploadUrl() /*-{
```

Comment:
I've read somewhere that the uploadUrl cannot be reused. 
The go documentation is mentioning that but not the java doc. Maybe it's not needed anymore ?